src/content_manager.cc

```diff
@@ -914,74 +915,107 @@ void ContentManager::updateObject(int objectID, const std::map<std::string, std:
     std::string location = getValueOrDefault(parameters, "location");
     std::string protocol = getValueOrDefault(parameters, "protocol");
 
-    auto obj = database->loadObject(objectID);
-    unsigned int objectType = obj->getObjectType();
-
-    if (IS_CDS_ITEM(objectType)) {
-        auto item = std::static_pointer_cast<CdsItem>(obj);
-        auto clone = CdsObject::createObject(objectType);
-        item->copyTo(clone);
-
-        if (!title.empty())
-            clone->setTitle(title);
-        if (!upnp_class.empty())
-            clone->setClass(upnp_class);
-        if (!location.empty())
-            clone->setLocation(location);
-
-        auto cloned_item = std::static_pointer_cast<CdsItem>(clone);
-
-        if (!mimetype.empty() && !protocol.empty()) {
-            cloned_item->setMimeType(mimetype);
-            auto resource = cloned_item->getResource(0);
-            resource->addAttribute(R_PROTOCOLINFO, renderProtocolInfo(mimetype, protocol));
-        } else if (mimetype.empty() && !protocol.empty()) {
-            auto resource = cloned_item->getResource(0);
-            resource->addAttribute(R_PROTOCOLINFO, renderProtocolInfo(cloned_item->getMimeType(), protocol));
-        } else if (!mimetype.empty()) {
-            cloned_item->setMimeType(mimetype);
-            auto resource = cloned_item->getResource(0);
-            std::vector<std::string> parts = splitString(resource->getAttribute(R_PROTOCOLINFO), ':');
-            protocol = parts[0];
-            resource->addAttribute(R_PROTOCOLINFO, renderProtocolInfo(mimetype, protocol));
-        }
+    log_error("updateCdsObject: CdsObject {} not updated", title);
```

Comment:
Perhaps this should throw an exception?

Reply:
it didn't before and I'm not sure whether the web page handler proberly deals with exceptions. The error message already is an improvement, if at any time in the future new classes are not handled.